app/server/lib/NSandbox.ts

```diff
@@ -129,52 +137,65 @@ export class NSandbox implements ISandbox {
     this._control = sandboxProcess.control;
     this.childProc = sandboxProcess.child;
 
-    this._logMeta = {sandboxPid: this.childProc.pid, ...options.logMeta};
+    this._logMeta = {sandboxPid: this.childProc?.pid, ...options.logMeta};
 
-    if (options.minimalPipeMode) {
-      log.rawDebug("3-pipe Sandbox started", this._logMeta);
-      if (sandboxProcess.dataToSandboxDescriptor) {
-        this._streamToSandbox =
-          (this.childProc.stdio as Stream[])[sandboxProcess.dataToSandboxDescriptor] as Writable;
-      } else {
-        this._streamToSandbox = this.childProc.stdin!;
-      }
-      if (sandboxProcess.dataFromSandboxDescriptor) {
-        this._streamFromSandbox =
-          (this.childProc.stdio as Stream[])[sandboxProcess.dataFromSandboxDescriptor];
+    if (this.childProc) {
```

Comment:
Any easy changes we can make here to simplify things? (e.g. splitting up into methods, lowering conditional depth)

Ok if not - just flagging because it was a bit of a struggle to read.

Reply:
I broke the constructor up somewhat.